circuits/cpp/barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -0,0 +1,291 @@
+        .y1 = y.get_witness_index(),
+        .x2 = other.x.get_witness_index(),
+        .y2 = other.y.get_witness_index(),
+        .x3 = result.x.get_witness_index(),
+        .y3 = result.y.get_witness_index(),
+        .endomorphism_coefficient = 1,
+        .sign_coefficient = 1,
+    };
+    context->create_ecc_add_gate(add_gate);
+
+    return result;
+}
+
+/**
+ * @brief will evaluate ECC point subtraction over `*this` and `other`.
+ *        Incomplete addition formula edge cases are *NOT* checked!
+ *        Only use this method if you know the x-coordinates of the operands cannot collide
+ *
+ * @tparam Composer
+ * @param other
+ * @return cycle_group<Composer>
+ */
+template <typename Composer>
+cycle_group<Composer> cycle_group<Composer>::unconditional_subtract(const cycle_group& other)
+{
+    auto context = get_context(other);
+
+    const bool lhs_constant = is_constant();
+    const bool rhs_constant = other.is_constant();
+
+    if (lhs_constant && !rhs_constant) {
+        auto lhs = cycle_group<Composer>::from_witness(context, get_value());
+        return lhs.unconditional_subtract(other);
+    }
+    if (!lhs_constant && rhs_constant) {
+        auto rhs = cycle_group<Composer>::from_witness(context, other.get_value());
+        return unconditional_subtract(rhs);
+    }
+    auto p1 = get_value();
+    auto p2 = other.get_value();
+    affine_element p3(element(p1) - element(p2));
+    if (lhs_constant && rhs_constant) {
+        return cycle_group(p3);
+    }
+    cycle_group result = cycle_group<Composer>::from_witness(context, p3);
+
+    proof_system::ecc_add_gate_<FF> add_gate{
+        .x1 = x.get_witness_index(),
+        .y1 = y.get_witness_index(),
+        .x2 = other.x.get_witness_index(),
+        .y2 = other.y.get_witness_index(),
+        .x3 = result.x.get_witness_index(),
+        .y3 = result.y.get_witness_index(),
+        .endomorphism_coefficient = 1,
+        .sign_coefficient = -1,
+    };
+    context->create_ecc_add_gate(add_gate);
+
+    return result;
+}
+
+/**
+ * @brief Will evaluate ECC point addition over `*this` and `other`.
+ *        Uses incomplete addition formula
+ *        If incomplete addition formula edge cases are triggered (x-coordinates of operands collide),
+ *        the constraints produced by this method will be unsatisfiable.
+ *        Useful when an honest prover will not produce a point collision with overwhelming probability,
+ *        but a cheating prover will be able to.
+ *
+ * @tparam Composer
+ * @param other
+ * @return cycle_group<Composer>
+ */
+template <typename Composer>
+cycle_group<Composer> cycle_group<Composer>::constrained_unconditional_add(const cycle_group& other)
+{
+    field_t x_delta = x - other.x;
+    x_delta.assert_is_not_zero("cycle_group::constrained_unconditional_add, x-coordinate collision");
+    return unconditional_add(other);
+}
+
+/**
+ * @brief Will evaluate ECC point subtraction over `*this` and `other`.
```

Comment:
General API suggestion: We could make the unchecked/unsafe versions of these harder to access by overriding the operators for the safe version and having methods for the unsafe versions.

So:  PointA - PointB would automatically do the safe version which includes the checks, while to do the unsafe version you need to go out of your way by doing PointA.unconditional_subtract(PointB)